llvm/lib/Transforms/Vectorize/VPlanPredicator.cpp

```diff
@@ -133,8 +133,8 @@ VPValue *VPPredicator::createBlockInMask(VPBasicBlock *VPBB) {
   // load/store/gather/scatter. Initialize BlockMask to no-mask.
   VPValue *BlockMask = nullptr;
   // This is the block mask. We OR all unique incoming edges.
-  for (auto *Predecessor : SetVector<VPBlockBase *>(
-           VPBB->getPredecessors().begin(), VPBB->getPredecessors().end())) {
+  for (auto *Predecessor : make_range(VPBB->getPredecessors().begin(),
+                                      unique(VPBB->getPredecessors()))) {
```

Comment:
Is this correct? It looks like unique only removes equivalent elements that are consecutive to each other.

Reply:
I think it also mutates the container?